.github/workflows/ci.yml

```diff
@@ -31,214 +34,42 @@ jobs:
-    env:
-      VERSION: ${{ github.sha }}
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - file: docker-compose.lan.yml
-            name: lan
-          - file: docker-compose.wan-hp.yml
-            name: wan-hp
-          - file: docker-compose.wan-relay.yml
-            name: wan-relay
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./.github/actions/gcp-docker-login
-        id: login
-        with:
-          project: firezone-staging
-      - name: Run ${{ matrix.file }} test
-        run: |
-          sudo sysctl -w vm.overcommit_memory=1
-          timeout 600 docker compose -f rust/snownet-tests/${{ matrix.file }} up --exit-code-from dialer --abort-on-container-exit
-
+  build-artifacts:
+    uses: ./.github/workflows/_build_artifacts.yml
+    secrets: inherit
   integration-tests:
-    name: integration-tests-${{ matrix.test }}
-    needs: build-images
-    runs-on: ubuntu-22.04
-    permissions:
-      contents: read
-      id-token: write
-      pull-requests: write
-    env:
-      VERSION: ${{ github.sha }}
-    strategy:
-      fail-fast: false
-      matrix:
-        test: [
-          direct-curl-portal-restart,
-          relayed-curl-portal-restart,
-          relayed-curl-relay-restart,
-          direct-curl-portal-down,
-          relayed-curl-portal-down,
-          direct-curl-portal-relay-down,
-          dns-etc-resolvconf,
-          dns-nm,
-          systemd/dns-systemd-resolved,
-        ]
-    steps:
-      - uses: actions/checkout@v4
-      - uses: ./.github/actions/gcp-docker-login
-        id: login
-        with:
-          project: firezone-staging
-      - name: Seed database
-        run: docker compose run elixir /bin/sh -c 'cd apps/domain && mix ecto.seed'
-      - name: Start docker compose in the background
-        run: |
-          # TODO: Order matters here, but it shouldn't. There seems to be some race
-          # condition involved in letting Docker deterime the start order here.
-          docker compose up -d dns.httpbin httpbin
-          docker compose up -d api web
-          docker compose up -d relay
-          docker compose up -d gateway
-          docker compose up -d client
-
-      - run: ./scripts/tests/${{ matrix.test }}.sh
+    uses: ./.github/workflows/_integration_tests.yml
+    needs: build-artifacts
+    secrets: inherit
 
-      - name: Show Client logs
-        if: "!cancelled()"
-        run: docker compose logs client
-      - name: Show Relay logs
-        if: "!cancelled()"
-        run: docker compose logs relay
-      - name: Show Gateway logs
-        if: "!cancelled()"
-        run: docker compose logs gateway
-      - name: Show API logs
-        if: "!cancelled()"
-        run: docker compose logs api
+  compatibility-tests:
+    # Allow breaking changes only if the PR contains a '!' in the title
+    if: github.event_name == 'pull_request' && ! contains(github.event.pull_request.title, '!')
```

Comment:
If this doesn't always run, we can't make it a required job, right?

I think I'd prefer if we have to use the "override with admin privileges" checkbox to merge a PR that has breaking changes.

Reply:
👍